packages/commonwealth/client/scripts/views/pages/finish_axie_login.tsx

```diff
@@ -34,9 +34,43 @@ const validate = async (
     return `Login Error: ${e.responseText}`;
   }
   if (result.status === 'Success') {
+    // Generate Canvas session key
+    const canvasMessage = {
+      loginTo: "/commonwealth",
+      registerSessionAddress: result.user.Addresses[0].address,
+      registerSessionDuration: 86400 * 1000,
+      timestamp: Math.floor((new Date()).getTime() / 1000)
+    };
+
+    // Ask ronin wallet to sign canvas message
+    const signature = await new Promise((resolve, reject) => {
+      // @ts-ignore
+      window.ronin.provider.sendAsync(
```

Comment:
So, uh, technically we're not supposed to interact directly with their wallet... we're using an OAuth flow on the backend.

Reply:
Oh interesting, what does "not supposed to" mean in this case? This function is mentioned in their documentation: https://docs.roninchain.com/docs/developer-guide/integrate-ronin-wallet